src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -309,25 +313,34 @@ module Buffer = struct
          List.rev !lines)
 end
 
-let special_file_for_fs_sync =
-  let path =
-    lazy
-      (let dir = Path.Build.relative Path.Build.root ".sync" in
-       Path.Build.relative dir "token")
-  in
-  fun () -> Lazy.force path
-
-let special_file_for_inotify_sync_absolute =
-  lazy (Path.to_absolute_filename (Path.build (special_file_for_fs_sync ())))
-
-let is_special_file_for_inotify_sync (path : Path.t) =
-  match path with
-  | In_source_tree _ -> false
-  | External _ ->
-    String.equal (Path.to_string path)
-      (Lazy.force special_file_for_inotify_sync_absolute)
-  | In_build_dir build_path ->
-    Path.Build.( = ) build_path (special_file_for_fs_sync ())
+let special_dir_for_fs_sync =
+  lazy
+    (Path.Build.relative Path.Build.root ".sync" |> Path.build |> Path.to_string)
+
+let emit_sync t =
+  let id = Sync_id.gen () in
+  let fn = id |> Sync_id.to_int |> string_of_int in
+  Table.set t.sync_table fn id;
+  let path = Filename.concat (Lazy.force special_dir_for_fs_sync) fn in
+  Unix.close (Unix.openfile path [ O_WRONLY; O_CREAT; O_TRUNC ] 0o666);
```

Comment:
Shouldn't you populate the table after the writing the file succeeds?

Reply:
Sounds good